dietpi/dietpi-software

```diff
@@ -3612,6 +3620,41 @@ _EOF_
 
 		fi
 
+		software_id=186 # IPFS Node
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Installing
+
+			local arch=
+			local version=
+			local fallback_url=
+
+			# - x86_64
+			if (( $G_HW_ARCH == 10 )); then
+
+				arch='amd64'
+
+			# - ARMv8
+			elif (( $G_HW_ARCH == 3 )); then
+
+				arch='arm64'
+
+			# - ARMv6/7
+			else
+				
+				arch='arm'
+
+			fi
+
+			fallback_url="https://dist.ipfs.io/go-ipfs/v0.8.0/go-ipfs_v0.8.0_linux-$arch.tar.gz"
+
+			version=$(curl -sSfL 'https://dist.ipfs.io/go-ipfs/versions' | tail -n 1)
+			
+			Download_Install "https://dist.ipfs.io/go-ipfs/$version/go-ipfs_${version}_linux-$arch.tar.gz"
```

Comment:
`fallback_url` does not work like that. It is only used when `$1` of `Download_Install` is an empty string, not as a general fallback. It is intended for that case the api.github.com "browser_download_url" scrapes return an empty string, because of changed download file names. Here it's similar to api.github.com "tag_name" scrapes, where we only get the version string and build the download URL with it:
```
local version=$(curl -sSfL 'https://dist.ipfs.io/go-ipfs/versions' | tail -n 1)
[[ $version ]] || { version='v0.8.0'; G_DIETPI-NOTIFY 1 "Automatic latest ${aSOFTWARE_NAME[$software_id]} version detection failed. Version \"$version\" will be installed as fallback, but a newer version might be available. Please report this at: https://github.com/MichaIng/DietPi/issues"; }
Download_Install "https://dist.ipfs.io/go-ipfs/$version/go-ipfs_${version}_linux-$arch.tar.gz"
```